### DISCUSSION TITLE
Using Oauth2 or digest auth doenst append the token to my actual url within the query

### DISCUSSION THREAD

**Discussion by dbdudubu** [grafana/grafana-infinity-datasource#1001] [Q&A]
Hi, I am working on using Oauth2.0 from esri or digest auth to request for access token which will use to append to my url in <url>?token=<access_token>. When I try the health check its working perfectly fine, but when I try to actual query, from query inspector I can see no token is being appended when using digest auth, while using Oauth2 with client credentails - > Authentication steps not included for OAuth authentication. 

Does anyone know how to request a token first, then parsing/append to url and fetch the API data?

**Comment by yesoreyeram** [comment]
Adding token to the the query string is not supported in standard oauth2 flow / infinity plugin yet.

**Reply by flux3all** [reply]
Confused - perhaps I am missing something. What is the point in having the OAuth2 authentication method implemented if the token is not sent with subsequent queries? Maybe this is a work in progress? thanks.

**Reply by yesoreyeram** [reply]
As per the OAuth2 spec, the token is sent via header not the query string. Sending token in subsequent requests via header is already supported

**Reply by flux3all** [reply]
Thank you for your quick response. Even though the Infiniti data source health check is successful, the query inspector output shown below does not show the token in the curl command as a header. 
Does "> Authentication steps not included for OAuth authentication" in the query inspector just mean to say that the token is not shown in the curl command?

Query inspector output:
###############
## URL
###############

https://query url
###############
## Curl Command
###############

curl -X 'GET' -H 'Accept: application/json;q=0.9,text/plain' 'https://query url'
###############
> Authentication steps not included for OAuth authentication
